docs/4.2. Behaviour - Clients.md

```diff
@@ -47,7 +47,10 @@ presence of the keys `registration_client_uri` and `registration_access_token` i
 Confidential Clients SHOULD support authentication using a digitally signed JSON Web Token, to authenticate with the
 Authorization Server's token and revocation endpoints in the manner described in Section 2.2 of [RFC 7523][RFC-7523].
 This authentication method is identified as `private_key_jwt` in the `token_endpoint_auth_method` string during client
-registration.
+registration. Clients SHOULD be capable of hosting their own public JWKs at a location that corresponds to the value of
+the registered `jwks_uri` used when registering with the Authorization Server. Clients are RECOMMENDED to use the
+`jwks_uri` in preference to `jwks` such that their public keys can be replaced without re-registering with the
+Authorization Server.
```

Comment:
If a node re-registered, if would presumably get a second client ID? If this is the case then a node would no longer be able to update the registry 🤔 Perhaps this may need to be a MUST?

Reply:
In my head I was considering if a client may choose to use https://tools.ietf.org/html/rfc7592 if the server implemented it in this case, but as it's only a MAY for the auth server to implement that you may be right that we're better going that way.

Reply:
https://tools.ietf.org/html/rfc7592#section-2.2 would be helpful. It appears `jwks_uri` is more common too perhaps for a reason related to this.

Reply:
Hopefully that's a little better now.